geomesa-convert/geomesa-convert-xml/src/main/scala/org/locationtech/geomesa/convert/xml/XmlConverter.scala

```diff
@@ -39,69 +39,23 @@ import scala.util.control.NonFatal
 class XmlConverter(sft: SimpleFeatureType, config: XmlConfig, fields: Seq[XmlField], options: XmlOptions)
     extends AbstractConverter[Element, XmlConfig, XmlField, XmlOptions](sft, config, fields, options) {
 
-  import scala.collection.JavaConverters._
-
-  private val docBuilder = {
-    val factory = DocumentBuilderFactory.newInstance()
-    factory.setNamespaceAware(true)
-    factory.newDocumentBuilder()
-  }
+  private val parser = new DocParser(config.xsd)
 
-  private val xmlValidator = config.xsd.map { path =>
-    val schemaFactory = SchemaFactory.newInstance(W3C_XML_SCHEMA_NS_URI)
-    WithClose(getClass.getClassLoader.getResourceAsStream(path)) { xsdStream =>
-      schemaFactory.newSchema(new StreamSource(xsdStream)).newValidator()
-    }
-  }
-
-  private val xpath = {
-    val factory = try {
-      val res = XPathFactory.newInstance(XPathFactory.DEFAULT_OBJECT_MODEL_URI, config.xpathFactory, getClass.getClassLoader)
-      logger.info(s"Loaded xpath factory ${res.getClass}")
-      res
-    } catch {
-      case NonFatal(e) =>
-        logger.warn(s"Unable to load xpath provider '${config.xpathFactory}': ${e.toString}. " +
-            "Xpath queries may be slower - check your classpath")
-        XPathFactory.newInstance(XPathFactory.DEFAULT_OBJECT_MODEL_URI)
-    }
-    val xp = factory.newXPath()
-    if (config.xmlNamespaces.nonEmpty) {
-      xp.setNamespaceContext(new NamespaceContext() {
-        override def getPrefix(namespaceURI: String): String = null
-        override def getPrefixes(namespaceURI: String): java.util.Iterator[_] = null
-        override def getNamespaceURI(prefix: String): String = config.xmlNamespaces.getOrElse(prefix, null)
-      })
-    }
-    xp
-  }
+  private val xpath = XmlConverter.createXPath(config.xpathFactory, config.xmlNamespaces)
 
   private val rootPath = config.featurePath.map(xpath.compile)
 
   fields.foreach(_.compile(xpath))
 
   // TODO GEOMESA-1039 more efficient InputStream processing for multi mode
```

Comment:
Does this comment need to move?

Reply:
ah, yeah it should probably go in the 'iterator' method, but i guess it's still relevant here where it's invoked...